warehouse/query-core/src/main/java/datawave/query/transformer/GroupingTransform.java

```diff
@@ -362,10 +362,31 @@ private void flatten(List<Document> documents) {
             if (this.groupFieldsSet.contains(shorterName)) {
                 expandedGroupFieldsList.add(shortName);
                 log.trace("{} contains {}", this.groupFieldsSet, shorterName);
-                GroupingTypeAttribute<?> created = makeGroupingTypeAttribute(shortName, field.getData());
-                created.setColumnVisibility(field.getColumnVisibility());
-                fieldMap.put(fieldName, created);
-                fieldToFieldWithContextMap.put(shortName, fieldName);
+                
+                if (field.getData() instanceof Collection<?>) {
+                    // This handles multi-valued entries that do not have gouping context
+                    // Create GroupingTypeAttribute and put in ordered map ordered on the attribute type
+                    Multimap<Type<?>,GroupingTypeAttribute<?>> attrSortedMap = TreeMultimap.create();
+                    for (Object typeAttribute : ((Collection<?>) field.getData())) {
+                        Type<?> type = ((TypeAttribute<?>) typeAttribute).getType();
```

Comment:
If we use `getData()` here instead of `getType()`, then it'll directly parallel the case below when we only have a single value. It took me a bit of digging to realize those two methods are actually the same so I was wondering why it'd be different for a multi-valued value vs a single value.

Reply:
getData() gets me an Object. At this point in the code I need a Type. That is why I used getType(). In the single value, all I have is an Attribute, so all I have is getData(), I think it should stay as is.